### ISSUE TITLE
Build error on OSX [commit 6b626af]

### ISSUE THREAD

**Issue by Vezzra** [freeorion/freeorion#588]
I get these errors when building GG:

```
In file included from /Users/user/SoftwareProjekte/FO/src/GG/src/GUI.cpp:44:
In file included from /Users/user/SoftwareProjekte/FO/src/GG/src/GIL/extension/io/png_io.hpp:41:
/Users/user/SoftwareProjekte/FO/src/GG/src/GIL/extension/io/png_io_private.hpp:284:69: error: no matching constructor for initialization of 'boost::gil::detail::png_reader'
    png_reader_color_convert(const boost::filesystem::path& path) : png_reader(path) {}
                                                                    ^          ~~~~
```

```
In file included from /Users/user/SoftwareProjekte/FO/src/GG/src/Texture.cpp:49:
In file included from /Users/user/SoftwareProjekte/FO/src/GG/src/GIL/extension/io/png_dynamic_io.hpp:37:
In file included from /Users/user/SoftwareProjekte/FO/src/GG/src/GIL/extension/io/png_io.hpp:41:
/Users/user/SoftwareProjekte/FO/src/GG/src/GIL/extension/io/png_io_private.hpp:284:69: error: no matching constructor for initialization of 'boost::gil::detail::png_reader'
    png_reader_color_convert(const boost::filesystem::path& path) : png_reader(path) {}
                                                                    ^          ~~~~
```

```
/Users/user/SoftwareProjekte/FO/src/GG/src/GIL/extension/io/png_io_private.hpp:370:21: error: cannot initialize object parameter of type 'boost::gil::detail::png_reader' with an expression of type 'boost::gil::detail::png_reader_color_convert<boost::gil::default_color_converter>'
        im.recreate(get_dimensions());
                    ^~~~~~~~~~~~~~
```

**Comment by geoffthemedio** [comment]
comment out line 284 in GG/src/GIL/extension/io/png_io_private.hpp

**Comment by Vezzra** [comment]
Yep, that worked. Well, that was quick :smiley: 

Shall I commit this fix?

**Comment by geoffthemedio** [comment]
sure

**Comment by Vezzra** [comment]
Fixed by d043ca275dc98127f399b890f6d2f8696ad05cc6
